ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -547,4 +591,72 @@ func bar;
         check("");
         check("         ");
     }
+
+    #[test]
+    fn const_int_strings() {
+        use num_traits::cast::NumCast;
+        use std::mem;
+
+        // Check (in an endian neutral manner) that a `num-bits`-sized integer of value `num`, when
+        // converted to a constant IR integer, then stringified, results in the string `expect`.
+        //
+        // When `num` has a bit size greater than `num_bits` the most significant bits of `num` are
+        // treated as undefined: they can be any value as IR stringification will ignore them.
+        fn check<T: NumCast + Sized>(num_bits: u32, num: T, expect: &str) {
+            dbg!(mem::size_of::<T>() * 8, usize::try_from(num_bits).unwrap());
+            assert!(mem::size_of::<T>() * 8 >= usize::try_from(num_bits).unwrap());
+
+            // Get a byte-vector for `num`.
+            let mut bytes: Vec<u8> = Vec::new();
+            match mem::size_of::<T>() {
+                1 => bytes
+                    .write_u8(<u8 as NumCast>::from::<T>(num).unwrap())
+                    .unwrap(),
+                2 => bytes
+                    .write_u16::<NativeEndian>(<u16 as NumCast>::from(num).unwrap())
+                    .unwrap(),
+                4 => bytes
+                    .write_u32::<NativeEndian>(<u32 as NumCast>::from(num).unwrap())
+                    .unwrap(),
+                8 => bytes
+                    .write_u64::<NativeEndian>(<u64 as NumCast>::from(num).unwrap())
+                    .unwrap(),
+                _ => panic!("{}", mem::size_of::<T>()),
```

Comment:
`todo` or `unreachable` ?

Reply:
I never know which to use.

This is something that we will need to implement one day, but is impossible under the current assumptions of the system. You tell me :P

Reply:
`todo` means "I will fix this later". `unreachable` means "if you've got here I've got something wrong". `panic` means "if you've got here, the caller has done something wrong".

Reply:
So it's a mix of `todo` and `unreachable`.

`unreachable` for now, `todo` in the future.

Inclined to go with `todo` so I don't have to (forget to) flip a `unreachable` to a `todo` later.

Reply:
Fixed in 27ece7c3abb3